plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/business/internal/migration/WorkspaceImageGMFBoundsMigrationParticipant.java

```diff
@@ -114,27 +105,41 @@ protected void postLoad(DAnalysis dAnalysis, Version loadedVersion) {
         }
     }
 
-    private boolean resizeWorkspaceImageGMFBounds(Object child, boolean previousMigrationApplied) {
-        boolean resized = false;
-        if (child instanceof Node && ((Node) child).getLayoutConstraint() instanceof Size) {
-            Node node = (Node) child;
-            if (node.getElement() instanceof DNode) {
-                DNode dnode = (DNode) node.getElement();
-                if (dnode.getStyle() != null) {
-                    StyleDescription description = dnode.getStyle().getDescription();
-                    resized = resizeGMFNode(previousMigrationApplied, node, description, dnode.getWidth(), dnode.getHeight());
-                }
-            } else if (node.getElement() instanceof DDiagramElementContainer) {
-                DDiagramElementContainer dDiagramElementContainer = (DDiagramElementContainer) node.getElement();
-                if (dDiagramElementContainer.getStyle() != null) {
-                    StyleDescription description = dDiagramElementContainer.getStyle().getDescription();
-                    resized = resizeGMFNode(previousMigrationApplied, node, description, dDiagramElementContainer.getWidth(), dDiagramElementContainer.getHeight());
-                    for (Object o : node.getChildren()) {
-                        resized = resizeWorkspaceImageGMFBounds(o, previousMigrationApplied);
-                    }
+    private boolean resizeWorkspaceImageGmfBoundsOfDiagram(boolean previousMigrationApplied, StringBuilder sb, DDiagram dDiagram, Diagram gmfDiagram) {
+        boolean migrationOccurredInCurrentDiag = false;
+        String representationName = StringUtil.EMPTY_STRING;
+        TreeIterator<EObject> childIterator = gmfDiagram.eAllContents();
+        while (childIterator.hasNext()) {
+            EObject eObject = childIterator.next();
+            if (eObject instanceof Node && ((Node) eObject).getLayoutConstraint() instanceof Size) {
+                Node node = (Node) eObject;
+                if (resizeWorkspaceImageGMFBounds(node, previousMigrationApplied)) {
+                    migrationOccurredInCurrentDiag = true;
+                    representationName = dDiagram.getName();
                 }
             }
         }
+        if (migrationOccurredInCurrentDiag) {
+            sb.append(MessageFormat.format(Messages.WorkspaceImageGMFBoundsMigrationParticipant_GMFBoundsResized, representationName));
+        }
+        return migrationOccurredInCurrentDiag;
+    }
+
+    private boolean resizeWorkspaceImageGMFBounds(Node node, boolean previousMigrationApplied) {
+        boolean resized = false;
+        if (node.getElement() instanceof DNode) {
+            DNode dnode = (DNode) node.getElement(); // 509, 204 KO and 484,186 OK
```

Comment:
What does `509, 204 KO and 484,186 OK` mean?

Reply:
Oops debug notes for myself. I'll remove them.